checks/cassandra/check.go

```diff
@@ -0,0 +1,45 @@
+package cassandra
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/gocql/gocql"
+)
+
+// Config is the Cassandra checker configuration settings container.
+type Config struct {
+	// Hosts is a list of Cassandra hosts. At least one is required.
+	Hosts []string
+	// Keyspace is the Cassandra keyspace to which you want to connect. Required.
+	Keyspace string
+}
+
+// New creates new Cassandra health check that verifies the following:
+// - that a connection can be established through creating a session
+// - that queries can be executed by describing keyspaces
+func New(config Config) func(ctx context.Context) error {
+	return func(ctx context.Context) (checkErr error) {
+		if len(config.Hosts) < 1 || len(config.Keyspace) < 1 {
+			checkErr = errors.New("keyspace name and hosts are required to initialize cassandra health check")
```

Comment:
looks like this error is not being returned and will likely get [overwritten later](https://github.com/gocql/gocql/blob/4367d090becb1a32ec585e539e70e62e5235594e/session.go#LL119C15-L119C25)

Reply:
Miss on my part. Thanks for catching it. Will update.